build-logic/src/main/kotlin/com.hedera.hashgraph.root.gradle.kts

```diff
@@ -14,3 +16,29 @@
  * ANY DAMAGES SUFFERED BY LICENSEE AS A RESULT OF USING, MODIFYING OR
  * DISTRIBUTING THIS SOFTWARE OR ITS DERIVATIVES.
  */
+
+plugins {
+    id("com.hedera.hashgraph.aggregate-reports")
+    id("com.hedera.hashgraph.dependency-analysis")
+    id("com.hedera.hashgraph.spotless-conventions")
+    id("com.hedera.hashgraph.spotless-kotlin-conventions")
+}
+
+repositories { mavenCentral() }
+
+tasks.register("checkAllModuleInfo")
+
+configureLifecycleTask("assemble")
+configureLifecycleTask("check")
+configureLifecycleTask("build")
+configureLifecycleTask("spotlessCheck")
+configureLifecycleTask("spotlessApply")
+configureLifecycleTask("checkAllModuleInfo")
+
+fun configureLifecycleTask(taskName: String) {
```

Comment:
This is needed for the umbrella build. Because builds are kept as independent as possible, even if they `includeBuild` each other, you normally can't do things like`./gradlew assemble` to run assemble on **all** projects. You have to explicitly make such _lifecycle tasks_ available and configure them. That is, you can decide which tasks in the subprojects (where the actual code is) will be triggered using `dependsOn` relationships between the tasks.

Which is done in `com.hedera.hashgraph.root` and `com.hedera.hashgraph.umbrella`

See: https://docs.gradle.org/current/userguide/structuring_software_products_details.html#using_an_umbrella_build